src/Controller/EntityCreateController.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\src\Controller\EntityCreateController.
+ */
+
+namespace Drupal\src\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Url;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Generic controller to provide /entity/add and /entity/add/{bundle} routes.
+ */
+class EntityCreateController extends ControllerBase {
+
+  /**
+   * Displays add custom entity links for available types.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request.
+   *
+   * @return array
+   *   A render array for a list of the custom entity types that can be added or
+   *   if there is only one custom entity type defined for the site, the function
+   *   returns the custom entity add page for that custom entity type.
+   */
+  public function addPage($entity_type_id, Request $request) {
+    $entity_type = $this->entityTypeManager()->getDefinition($entity_type_id);
+    // Get the storage controller for this entity.
+    $bundle_storage = $entity_type->getBundleEntityType()
```

Comment:
The ternary operator usage here (and in this class in general) really impacts readability.